### PR TITLE
go-controller: informer factory refactor and minor cleanups

### PR DESCRIPTION
**Pull request by dcbw**
Make the factory more of a watcher factory, instead of something that creates the OVN and Cluster controllers itself.  Lets us make the controllers more self-contained and unit-testable in the future.

@rajatchopra @shettyg

### CODE REVIEW

**Comment by dcbw** [comment]
@shettyg any thoughts on this refactor?

**Comment by shettyg** [comment]
@dcbw 

I haven't tested your changes yet. I will give it a run. Overall it looks good to me. @rajatchopra Would you mind looking at the factory code again - you have more inside knowledge on any pitfalls.

**Comment by dcbw** [comment]
> I haven't tested your changes yet. I will give it a run

@shettyg thanks, really looking forward to Rajat's CI work so we can be more confident about this kind of stuff before merging.

**Comment by shettyg** [comment]
Sanity testing passed. Go from my side.

File: go-controller/pkg/factory/factory.go [hunk 1]
```diff
@@ -1,118 +1,93 @@
...
+		iFactory:  iFactory,
+		informers: make(map[string]cache.SharedIndexInformer),
+	}
+
+	wf.informers[typePods] = iFactory.Core().V1().Pods().Informer()
+	wf.informers[typeServices] = iFactory.Core().V1().Services().Informer()
+	wf.informers[typeEndpoints] = iFactory.Core().V1().Endpoints().Informer()
+	wf.informers[typePolicies] = iFactory.Networking().V1().NetworkPolicies().Informer()
+	wf.informers[typeNamespaces] = iFactory.Core().V1().Namespaces().Informer()
+	wf.informers[typeNodes] = iFactory.Core().V1().Nodes().Informer()
```
Comment (rajatchopra) [inline go-controller/pkg/factory/factory.go]: The work looks good to me. At some point if we were to do selective watching, we will have to introduce something similar to the earlier method, though. Maybe just pass flags on which informers to initialize and which ones to skip. Maybe tie this function to the Run() function at that point.

**Pull request merged**